Iceberg: varbinary type and bucket partition trnsformation

Problem reproduction:

**Step 1:** `CREATE TABLE t(col_1 VARBINARY) WITH (partitioning=array['bucket(col_1, 16)'])`
**Step 2:** `INSERT INTO t(col_1) VALUES (x'01')`
**Result:** `throw (io.trino.plugin.iceberg.IcebergPageSink#applyTransform) ClassCastException (message: class [B cannot be cast to class java.nio.ByteBuffer)`

<details>
<summary><b>Stack trace</b></summary>

```java
Caused by: java.lang.ClassCastException: class [B cannot be cast to class java.nio.ByteBuffer ([B and java.nio.ByteBuffer are in module java.base of loader 'bootstrap')
	at org.apache.iceberg.transforms.Bucket$BucketByteBuffer.apply(Bucket.java:230)
	at io.trino.plugin.iceberg.IcebergPageSink.applyTransform(IcebergPageSink.java:471)
	at io.trino.plugin.iceberg.IcebergPageSink.getPartitionData(IcebergPageSink.java:463)
	at io.trino.plugin.iceberg.IcebergPageSink.getWriterIndexes(IcebergPageSink.java:347)
	at io.trino.plugin.iceberg.IcebergPageSink.writePage(IcebergPageSink.java:273)
	at io.trino.plugin.iceberg.IcebergPageSink.doAppend(IcebergPageSink.java:267)
	at io.trino.plugin.iceberg.IcebergPageSink.appendPage(IcebergPageSink.java:220)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorPageSink.appendPage(ClassLoaderSafeConnectorPageSink.java:68)
	at io.trino.operator.TableWriterOperator.addInput(TableWriterOperator.java:276)
	at io.trino.operator.Driver.processInternal(Driver.java:408)
	at io.trino.operator.Driver.lambda$process$8(Driver.java:306)
	at io.trino.operator.Driver.tryWithLock(Driver.java:709)
	at io.trino.operator.Driver.process(Driver.java:298)
	at io.trino.operator.Driver.processForDuration(Driver.java:269)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:890)
	at io.trino.execution.executor.dedicated.SplitProcessor.run(SplitProcessor.java:77)
	at io.trino.execution.executor.dedicated.TaskEntry$VersionEmbedderBridge.lambda$run$0(TaskEntry.java:191)
	at io.trino.$gen.Trino_testversion____20240522_133059_1.run(Unknown Source)
	at io.trino.execution.executor.dedicated.TaskEntry$VersionEmbedderBridge.run(TaskEntry.java:192)
	at io.trino.execution.executor.scheduler.FairScheduler.runTask(FairScheduler.java:168)
	at io.trino.execution.executor.scheduler.FairScheduler.lambda$submit$0(FairScheduler.java:155)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:76)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
```

</details>

--- a/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java
+++ b/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java
@@ -513,7 +513,7 @@ public class IcebergPageSink
             return timestampTzToMicros(getTimestampTz(block, position));
         }
         if (type instanceof VarbinaryType varbinaryType) {
-            return varbinaryType.getSlice(block, position).getBytes();
+            return varbinaryType.getSlice(block, position).toByteBuffer();
         }
         if (type instanceof VarcharType varcharType) {
             return varcharType.getSlice(block, position).toStringUtf8();

--- a/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java
+++ b/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java
@@ -3236,6 +3236,7 @@ public abstract class BaseIcebergConnectorTest
                 "CAST('206caec7-68b9-4778-81b2-a12ece70c8b1' AS UUID)",
                 "CAST('906caec7-68b9-4778-81b2-a12ece70c8b1' AS UUID)",
                 "CAST('406caec7-68b9-4778-81b2-a12ece70c8b1' AS UUID)");
+        testBucketTransformForType("VARBINARY", "x'04'", "x'21'", "x'02'");
     }
 
     protected void testBucketTransformForType(

